src/services/message.service.ts

```diff
@@ -39,6 +39,34 @@ export class MessageService {
     }
   }
 
+  async sendPagedEmbed(message: IMessage, pages: MessageEmbed[]): Promise<IMessage> {
+    const msg: IMessage = await message.channel.send(pages[0]);
+
+    msg.react('➡️');
+    const collecter = msg.createReactionCollector(
+      (reaction: MessageReaction, user: User) =>
+        ['⬅️', '➡️'].includes(reaction.emoji.name) && user.id !== msg.author.id, //Only run if its not the bot putting reacts
+      { time: 1000 * 60 * 10 } //Listen for 10 Minutes
+    );
+
+    let pageIndex = 0;
+    collecter.on('collect', (reaction: MessageReaction) => {
+      msg.reactions.removeAll().then(async () => {
+        reaction.emoji.name === '➡️' ? pageIndex++ : pageIndex--;
+        msg.edit(pages[pageIndex]);
```

Comment:
can be out of bounds

Reply:
It only sends the arrows if its [available.](https://github.com/joey-colon/lion/blob/65a5836da2702a98008c3c1c051ba068bbd1dd82/src/services/message.service.ts#L58-L63)
So the user cannot go past the bounds

Reply:
nice sneaky invariant. what happens if two people react at the same time though?

Reply:
in a way such that you could increment past the pages' bounds